From: Iain Lane <laney@debian.org>
Date: Mon, 18 Nov 2019 11:09:13 +0000
Subject: test_copy_preserve_mode: Adjust for revert semantics

Now we're returning the file type again, we need to mask it out to
compare with the mode.

Bug-Upstream: https://gitlab.gnome.org/GNOME/glib/issues/1934
---
 gio/tests/file.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/tests/file.c b/gio/tests/file.c
index 33dc50a..71dc85f 100644
--- a/gio/tests/file.c
+++ b/gio/tests/file.c
@@ -927,7 +927,7 @@ test_copy_preserve_mode (void)
                                      NULL, &local_error);
       g_assert_no_error (local_error);
 
-      dest_mode = g_file_info_get_attribute_uint32 (dest_info, G_FILE_ATTRIBUTE_UNIX_MODE);
+      dest_mode = g_file_info_get_attribute_uint32 (dest_info, G_FILE_ATTRIBUTE_UNIX_MODE) & ~S_IFMT;
 
       g_assert_cmpint (dest_mode, ==, vectors[i].expected_destination_mode);
 
